.github/workflows/development.yml

```diff
@@ -0,0 +1,43 @@
+name: Development Build and Heroku Push
+
+on:
+  push:
+    branches: [ development ]
+  pull_request:
+    branches: [ development ]
```

Comment:
One problem with this current setup is that your tests and deployment are orthogonal - failed tests won't stop deployments from being released. The simple solution would be to combine these workflows, you could have them as two separate "jobs" and add something like:
```yml
    # make the second job rely on the first one, assuming the first is called test
    needs: test
    # And this line will ensure that the second job only runs if the conditions are met, so the workflow level `on` conditions don't always trigger a deployment
   if: github.ref == 'refs/heads/main' && github.event_name != 'pull_request' 
```

If you want to take advantage of the code-cleanliness of keeping them separate, we could go with [reusable workflows](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#creating-a-reusable-workflow), and having this workflow call the testing workflow before deployment occurs - fair notice that I haven't tried using reusable workflows though!